contrib/osc.zsh

```diff
@@ -0,0 +1,149 @@
+#compdef osc
+#
+# Copyright (C) 2009,2010 Holger Macht <holger@homac.de>
+#
+# This file is released under the GPLv2.
+#
+# Based on the zsh guide from http://zsh.dotsrc.org/Guide/zshguide06.html
+#
+# Toggle verbose completions: zstyle ':completion:*:osc:*' verbose no
+#                             zstyle ':completion:*:osc-subcommand:*' verbose no
+# 
+# Use the variables $ZSH_OSC_BUILD_TARGETS_EXTRA and $ZSH_OSC_PROJECTS_EXTRA to
+# extend the list of possible completions in your ~/.zshrc like that:
+#  export OSC_PROJECTS_EXTRA="Base:System Base:shells"
+#
+# version 0.2
+#
+
+OSC_BUILD_TARGETS="openSUSE_13.1 openSUSE_13.2 openSUSE_Tumbleweed openSUSE_Factory SLE_11_SP3 SLE_12"
+OSC_PROJECTS="openSUSE:Factory openSUSE:Tumbleweed openSUSE:13.2 openSUSE:13.1"
```

Comment:
I don't like the hard-coded lists.
On the other hand, I don't see any other reasonable alternative at the moment.
Could you update the values to the active repos and projects at least?

Reply:
> Sure I can do that. The bash completition uses `~/.osc.projects` instead maybe something like this works but also the path is wrong as it is not inside `$XDG_CACHE_HOME`.

Yes, that needs to be fixed. I've submitted https://github.com/openSUSE/osc/pull/1364

Reply:
Projects are no longer depended on hardcoded list. Now on targets and arches are left.

Reply:
Now targets are done.

Reply:
Architectures are now fetched from the project repositories too. Should be all good now too.